src/test/rpc_wallet_tests.cpp

```diff
@@ -327,6 +327,39 @@ BOOST_AUTO_TEST_CASE(rpc_wallet_getbalance)
     BOOST_CHECK_THROW(CallRPC("z_listreceivedbyaddress tnRZ8bPq2pff3xBWhTJhNkVUkm2uhzksDeW5PvEa7aFKGT9Qi3YgTALZfjaY4jU3HLVKBtHdSXxoPoLA3naMPcHBcY88FcF 1"), runtime_error);
 }
 
+/**
+ * This test covers RPC command z_validateviewingkey
+ */
+BOOST_AUTO_TEST_CASE(rpc_wallet_z_validateviewingkey)
+{
+    SelectParams(CBaseChainParams::MAIN);
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    UniValue retValue;
+
+    // Check number of args
+    BOOST_CHECK_THROW(CallRPC("z_validateviewingkey"), runtime_error);
+    BOOST_CHECK_THROW(CallRPC("z_validateviewingkey toomany args"), runtime_error);
+
+    BOOST_CHECK_NO_THROW(retValue = CallRPC("z_validateviewingkey VKstuff"));
+    UniValue resultObj = retValue.get_obj();
+    bool b = find_value(resultObj, "isvalid").get_bool();
+    BOOST_CHECK_EQUAL(b, false);
+
+    BOOST_CHECK_NO_THROW(retValue = CallRPC("z_validateviewingkey ZiVKp4WNCjCsE8f1gAMySa6d8SmixgoErWnnuZG7wAVTDV3G9K5NP5gFNji61M4rTiVYhSwW4tnjF6vocC1HiN7WgkihUiyXZ"));
+    resultObj = retValue.get_obj();
+
+    b = find_value(resultObj, "isvalid").get_bool();
+    BOOST_CHECK_EQUAL(b, true);
+
+    b = find_value(resultObj, "isvalid").get_bool();
```

Comment:
This code is the same as above, copy and paste error?

Reply:
@leto We might be able to get this in for v1.1.2 ; merge window closing Friday.  Can you check the duplicate code above?  Thanks.

Reply:
@bitcartel I am working on this now and will have an update shortly